src/librustc_typeck/check/_match.rs

```diff
@@ -455,7 +528,11 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
         //     assert_eq!(foo.0.0, 42);
         // }
         // ```
-
+        //
+        // FIXME(tschottdorf): don't call contains_explicit_ref_binding, which
+        // is problematic as the HIR is being scraped, but ref bindings may be
+        // implicit after #42640. We need to make sure that pat_adjustments
+        // (once introduced) is populated by the time we get here.
```

Comment:
This is out of cache for me again. I think I was convinced that this was ok, but I have to double check. =) Didn't we discuss this *somewhere*?

Reply:
https://github.com/rust-lang/rust/pull/43399#discussion_r130223956